### DISCUSSION TITLE
updateLockFiles=false but Gradle still executed with --update-locks

### DISCUSSION THREAD

**Discussion by pkubowicz** [renovatebot/renovate#29486] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, 37.385.0

### Please tell us more about your question or problem

I have `updateLockFiles: false` in my config, but Renovate still executes `./gradlew ... --update-locks`. To me, it looks like `updateLockFiles` has no effect on Gradle integration.

I minimized my config to a fairly simple file:
```js
module.exports = {
  extends: ["config:recommended"],
  platform: "gitlab",
  gitPrivateKey: process.env.RENOVATE_GPG_KEY,
  secrets: {
  // my secrets
  },
  customEnvVariables: {
    // my var
  },
  // Disable unused package managers
  dockerfile: {
    enabled: false,
  },
  updateLockFiles: false,
};
```

### Need for disabling
In case you ask why I want to disable updating locks when my repository contains `gradle.lockfile`.

I am facing a problem where Renovate updates just `compileClasspath` in the lockfile (leaving all other configurations on the old version). It's hard to diagnose:
* running same Renovate configuration on a small Gradle project with reasonably similar configuration -> lock update correct
* copy-pasting `./gradlew` call from Renovate debug logs and executing on the complicated project but locally -> lock update correct
* adding postUpgradeTasks with brute-force `./gradlew dependencies --write-locks` -> lock incorrect

I have no idea how to reduce the problem to something that can be shared in a bug report (either to you or to Gradle). I'm not able to understand what happens to Gradle that when executed in Renovate runner, it gives different results for same commands.

So I wanted to stop Renovate from running Gradle altogether and write my own Gradle calls in postUpgradeTasks. But I'm not able to achieve this using `updateLockFiles: false`.

### Logs (if relevant)

<details><summary>Logs</summary>

```
DEBUG: Combined config
       "config": {
         "extends": ["config:recommended"],
         "platform": "gitlab",
         "gitPrivateKey": "***********",
         "secrets": {
         },
         "customEnvVariables": {
         },
         "dockerfile": {"enabled": false},
         "terraform": {"enabled": false},
         "gitlabci": {"enabled": false},
         "gitlabci-include": {"enabled": false},
         "docker-compose": {"enabled": false},
         "updateLockFiles": false,
         "hostRules": [
           {"hostType": "github", "matchHost": "github.com", "token": "***********"}
         ],
         "repositoryCache": "enabled",
         "baseDir": "***",
         "logFile": "renovate-log.ndjson",
         "logFileLevel": "debug",
         "onboardingConfig": {
           "$schema": "https://docs.renovatebot.com/renovate-schema.json",
           "extends": ["config:recommended"]
         },
         "optimizeForDisabled": true,
         "endpoint": "https://gitlab.com/api/v4",
         "token": "***********",
         "repositories": ["superside/platform/gradle-lock"]
       }

DEBUG: Updating found Gradle dependency lockfiles (repository=superside/platform/gradle-lock, branch=renovate/langversion)

DEBUG: Executing command (repository=superside/platform/gradle-lock, branch=renovate/langversion)
       "command": "./gradlew --console=plain --dependency-verification lenient -q properties"

DEBUG: Executing command (repository=superside/platform/gradle-lock, branch=renovate/langversion)
       "command": "install-tool java 17.0.11+9"
DEBUG: exec completed (repository=superside/platform/gradle-lock, branch=renovate/langversion)
       "durationMs": 243,
       "stdout": "",
       "stderr": ""
DEBUG: Executing command (repository=superside/platform/gradle-lock, branch=renovate/langversion)
       "command": "./gradlew --console=plain --dependency-verification lenient -q :dependencies :project-bom:dependencies --update-locks org.apache.commons:commons-lang3"

```

</details>

**Comment by rarkins** [comment]
I checked the source code and it's npm-only still. Docs fix: https://github.com/renovatebot/renovate/pull/29488
